h2o-persist-gcs/src/main/java/water/persist/PersistGcs.java

```diff
@@ -0,0 +1,161 @@
+package water.persist;
+
+import com.google.cloud.ReadChannel;
+import com.google.cloud.storage.*;
+import water.H2O;
+import water.Key;
+import water.Value;
+import water.api.FSIOException;
+import water.fvec.GcsFileVec;
+import water.util.Log;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Persistence backend for GCS
+ */
+public final class PersistGcs extends Persist {
+
+  private static final Storage storage = StorageOptions.getDefaultInstance().getService();
+
+  @Override
+  public byte[] load(Value v) throws IOException {
+    final BlobId blobId = GcsBlob.of(v._key).getBlobId();
```

Comment:
After import_files() finishes, this method is called multiple times for a single file, downloding and parsing it multiple times.

A simple Python script like this can reproduce the behavior:

```python
    parquet = h2o.import_file(path="gs://gcp-public-data-nexrad-l2/2018/01/01/KABR/NWS_NEXRAD_NXL2DPBL_KABR_20180101050000_20180101055959.tar")
    frame = parquet.as_data_frame()
    
    assert frame is not None;
```